### PR TITLE
Remove authorization header on redirect to different host

### LINKED ISSUE #1: remove authorization header on redirect to different host
`request` removes the authorization header while redirecting if the host changes: https://github.com/request/request/blob/b12a6245d9acdb1e13c6486d427801e123fdafae/lib/redirect.js#L134-L138

node-fetch is still sending all the headers, which is causing https://github.com/zkat/pacote/issues/87

So, we should remove that header too :)

### PR DESCRIPTION
**Pull request by ivanph**
Matches the behavior of request:
https://github.com/request/request/blob/b12a6245/lib/redirect.js#L134-L138

Fixes #1 

A few cases are missing from the tests but I wasn't sure if I should create a new server just for that and I dunno if I should test all redirect status codes for this case.

I wasn't sure if this had to be handled in that piece of code you pointed to, cause that only seems to cover POST requests and from the `request` code it seems that they do it for all methods but HEAD

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -86,6 +86,12 @@ function fetch (uri, opts) {
           reject(new FetchError(`redirect location header missing at: ${request.url}`, 'invalid-redirect'))
           return
         }
+        // Remove authorization if changing hostnames (but not if just
+        // changing ports or protocols).  This matches the behavior of request:
+        // https://github.com/request/request/blob/b12a6245/lib/redirect.js#L134-L138
+        if (url.parse(request.url).hostname !== url.parse(res.headers.location).hostname) {
```
Comment (zkat) [inline src/index.js]: I think we need to emulate request's behavior a little more closely here: `res.headers.location` can be `/foo` or `foo.com/x`, which would not have a `hostname` property when put through `url.parse()`.

It might be nicer to have code that's more explicit about detecting these different cases, instead of using request's concise but not-very-explicit implementation of it.

This patch looks otherwise fine, and I'm mostly fine with the tests as they are, modulo adding tests for `Location: /foo` and `Location: foo.com` and such

**Review by zkat — CHANGES_REQUESTED** [review]
This looks great so far! Thanks so much for picking it up! I made a comment regarding some corner cases with the Location header, but things otherwise look fine 🎉

File: src/index.js (hunk 1 above)
Comment (zkat) [reply src/index.js]: Correction!

The [`Location` header](https://tools.ietf.org/html/rfc7231#section-7.1.2) accepts a [`URI-Reference`](https://tools.ietf.org/html/rfc3986#section-4.1). That is, _either_ a relative ref, or a fully-qualified URI (with ports). `www.foo.com` is invalid. I kinda wonder if anyone sends it that way, though.

Comment (zkat) [reply src/index.js]: Correction again: It [allows _protocol-relative URIs_](https://tools.ietf.org/html/rfc3986#section-4.2) (that is, `//foo.com/bar`). So that still needs to be compensated for. Heh.

**Comment by ivanph** [comment]
Thanks!! I'll try to make this changes ASAP

File: src/index.js (hunk 1 above)
Comment (ivanph) [reply src/index.js]: I think I solved the issues with parsing different types of urls and also added a couple more test cases. I wasn't sure about URIs like `foo.com` cause it seems those are not valid but maybe I'm misreading.  Lemme know if there is anymore changes I should make.

**Review by zkat — APPROVED** [review]
LGTM! Great work :D

**Comment by ivanph** [comment]
Awesome!! 
Thanks for all the great work you've been doing on npm
